modules/comments/comments.php

```diff
@@ -465,13 +476,17 @@ public function pre_comment_on_post( $comment ) {
 			$post_array['hc_avatar'] = htmlentities( $post_array['hc_avatar'] );
 		}
 
-		$check = Jetpack_Comments::sign_remote_comment_parameters( $post_array, Jetpack_Options::get_option( 'blog_token' ) );
+		$blog_token = Jetpack_Data::get_access_token( false, $post_array['token_key'] );
+		if ( ! $blog_token ) {
+			wp_die( __( 'Unknown security token.', 'jetpack' ), 400 );
```

Comment:
I seem to get stuck here in my tests. `$blog_token` returns `false` because the `hash_equals` check in `class.jetpack-data.php` fails.